### PR TITLE
Storm commands for triggers and cron jobs

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/cmds/cron.py [hunk 1]
```diff
@@ -571,6 +571,9 @@ def _format_timestamp(ts):
                 self.printf(f'                 {incunit:10} {incval:6} {reqdict}')
 
     async def runCmdOpts(self, opts):
+        self.printf('Warning: Cron jobs should now be accessed via storm commands. '
```
Comment (vEpiphyte) [inline synapse/cmds/cron.py]: Can we please use the the warnings library to handle warnings as opposed to printing data?

File: synapse/cmds/trigger.py [hunk 2]
```diff
@@ -279,6 +279,9 @@ def _make_argparser(self):
         self.printf(f'Disabled trigger {iden}')
 
     async def runCmdOpts(self, opts):
+        self.printf('Warning: Triggers should now be accessed via storm commands. '
```
Comment (vEpiphyte) [inline synapse/cmds/trigger.py]: warnings

File: synapse/cortex.py [hunk 3]
```diff
@@ -3348,6 +3350,103 @@ def _loadCoreModule(self, ctor, conf=None):
...
+        crons = []
+
+        for iden, cron in self.agenda.list():
+            info = cron.pack()
+            info['username'] = self.getUserName(cron.useriden)
+            crons.append((iden, info))
+
+        return crons
+
+
```
Comment (vEpiphyte) [inline synapse/cortex.py]: extra whitespace

File: synapse/cortex.py [hunk 4]
```diff
@@ -3348,6 +3350,103 @@ def _loadCoreModule(self, ctor, conf=None):
         '''
         return await self.view.listTriggers()
 
+    @staticmethod
+    def _convert_reqdict(reqdict):
+        return {s_agenda.TimeUnit.fromString(k): v for (k, v) in reqdict.items()}
+
+    async def addCronJob(self, user, query, reqs, incunit=None, incval=1):
```
Comment (vEpiphyte) [inline synapse/cortex.py]: With these new methods, CoreAPI already implements many of them. Can you make the coreapi call the Cell implementation so we're not repeating ourselves ?  Future selves will thank us.

Comment (vEpiphyte) [reply synapse/cortex.py]: * Can you make the coreapi call the Cell implementation * with the prerequisite permission checks of course :)

Comment (vEpiphyte) [reply synapse/cortex.py]: Also, is it intended that we will remove the CoreApi imlementations in 0.3.0 ? if so those would need deprecation warnings as well.

@jnwatson

File: synapse/tests/test_utils.py [hunk 5]
```diff
@@ -203,3 +203,6 @@ def test_istufo(self):
 
             mesgs = await core.streamstorm('iden newp').list()
             self.stormIsInWarn('Failed to decode iden', mesgs)
+
+            mesgs = await core.streamstorm('[test:str=').list()
+            self.stormIsInErr('Unexpected end', mesgs)
```
Comment (vEpiphyte) [inline synapse/tests/test_utils.py]: https://www.youtube.com/watch?v=n__1Y-N5tQk

File: synapse/tests/utils.py [hunk 6]
```diff
@@ -1543,6 +1543,17 @@ def stormIsInWarn(self, mesg, mesgs):
         print_str = '\n'.join([m[1].get('mesg') for m in mesgs if m[0] == 'warn'])
         self.isin(mesg, print_str)
 
+    def stormIsInErr(self, mesg, mesgs):
```
Comment (vEpiphyte) [inline synapse/tests/utils.py]: :heart_eyes:

File: synapse/cortex.py (hunk 4 above)
Comment (Cisphyx) [reply synapse/cortex.py]: CoreAPI updated to call the Cell implementations.

**Pull request merged**